### ISSUE TITLE
Feature request: Support include request body for HTTP DELETE method.

### ISSUE THREAD

**Issue by gnoot** [Alamofire/Alamofire#411]
There are some API that allow to use request body for HTTP DELETE.
( e.g. http://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html )

if you support this, I'm glad allow to use request parameter and request body at the same time.

if it is already implemented, please tell me.:smile:

**Comment by mattt** [comment]
The HTTP specification does not support a message body for `DELETE` requests. As such, the default behavior is to append parameters into the URL query. 

If your API insists on this convention, you can construct the request yourself.

**Comment by kylef** [comment]
It's not unsupported, it just has no defined rules. An implementation may define semantics on top of HTTP such as this one.

From RFC 7231:

> A payload within a DELETE request message has no defined semantics;
> sending a payload body on a DELETE request might cause some existing
> implementations to reject the request.

**Comment by mattt** [comment]
Ah, so they finally clarified that for RFC 7231. TIL.

**Comment by aliuncoBamilo** [comment]
So how can I construct a request which let me to send body for DELETE http requests?  😞

**Comment by fmatosic** [comment]
Does Alamofire have support for body content in DELETE http request?

**Comment by jshier** [comment]
Sure, using the `JSONEncoding`, if you need the parameters JSON encoded. Otherwise you'll need to provide your own encoder.

**Comment by fmatosic** [comment]
thnx, using manager so this worked for me:
manager.request(lUrl, method: .delete, parameters: result, encoding: URLEncoding.httpBody, headers: headerss)
            .validate(contentType: ["application/json"])

**Comment by sw-tt-krutikvora** [comment]
i am facing issue in .delete call with parameters array (not Dictionary) .  and i getting request time out error. every time.
Server execute the request but Almofire can't able to get response so it automatically set status code 444

Please help me out

**Comment by ankit0812** [comment]
> i am facing issue in .delete call with parameters array (not Dictionary) . and i getting request time out error. every time.
> Server execute the request but Almofire can't able to get response so it automatically set status code 444
> 
> Please help me out

Where you able to solve this?

**Comment by JeanRomain** [comment]
I have this problem, if I set the method to DELETE , specify parameters then debugPrint the request, the body shows as empty in the debugPrint
If I do the same using an JSONEncoder the correct body shows in the debugPrint.

But in this case the server still don't receive the body and return the same error message.

I saw somewhere that in Swift 5 urlresponse can't handle body in DELETE request.
Does that affect alamofire? Is there a workaround (or other swift modules)?
